test/README.MD

```diff
@@ -34,6 +34,22 @@ long float a;
 enum Foo {};
 ```
 ---
+## Testing type resolution
+Type resolution can be tested by defining a `EXPECTED_TYPES` macro, and a function.
+The nth token in the `EXPECTED_TYPES` macro is the expected type of the nth statement
+in the function
+
+**example:**
+```c
+void my_test(void) {
+    int x = 5;
+    1L;
+    (void)1L;
+}
+#define EXPECTED_TYPES \
```

Comment:
This seems like a decent solution for now but eventually we'll wan to switch to using `__builtin_types_compatible` and similiar functions combined with static asserts.